bin/crew

```diff
@@ -1,16 +1,34 @@
 #!/usr/bin/env ruby
-require 'uri'
-require 'digest/sha2'
-begin
-  require 'json'
-rescue LoadError
-  # The json gem can break when upgrading from a much older version of ruby.
-  puts ' -> install gem json'
-  Gem.install('json')
-  require 'json'
+require_relative '../lib/color'
+# Disallow sudo
+abort 'Chromebrew should not be run as root.'.lightred if Process.uid.zero?
+def require_gem(gem_name_and_require = nil, require_override = nil)
+  # Allow only loading gems when needed.
+  return if gem_name_and_require.nil?
+
+  gem_name = gem_name_and_require.split('/')[0]
+  begin
+    gem gem_name
+  rescue LoadError
+    puts " -> install #{gem_name} gem".orange
+    Gem.install(gem_name)
+    gem gem_name
+  end
+  requires = if require_override.nil?
+               gem_name_and_require.split('/')[1].nil? ? gem_name_and_require.split('/')[0] : gem_name_and_require
+             else
+               require_override
+             end
+  require requires
 end
```

Comment:
This def block can be below all the require* lines, correct?

Reply:
It's up there on purpose. The color require is needed in that def block for `require_gem`, and `require_gem` enables the three gems below that, which are needed to enable debugging shortly below where the debugging enabledment is right after `const.rb` is loaded.

Reply:
The goal is to allow exit as soon as possible if run with sudo... ;)

Reply:
Fair enough.